tests/foreman/api/test_notifications.py

```diff
@@ -362,3 +392,48 @@ def test_positive_notification_recipients(target_sat):
     recipients = target_sat.api.NotificationRecipients().read()
     for notification in recipients.notifications:
         assert set(notification_keys) == set(notification.keys())
+
+
+@pytest.mark.tier3
+@pytest.mark.parametrize(
+    'admin_user_with_custom_settings',
+    [
+        pytest.param({'disabled': True, 'mail_enabled': True}, id='account_disabled'),
+        pytest.param({'disabled': False, 'mail_enabled': False}, id='mail_disabled'),
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures(
+    'reschedule_long_running_tasks_notification',
+    'wait_for_no_long_running_task_mail',
+)
+def test_negative_no_notification_for_long_running_tasks(
+    admin_user_with_custom_settings, long_running_task, root_mailbox_copy
+):
+    """Check that an e-mail notification for a long-running task
+    (i.e., running or paused for more than two days)
+    is NOT sent to users with disabled account or disabled e-mail.
+
+    :id: 03b41216-f39b-11ee-b9ea-000c2989e153
+
+    :setup:
+        1. Create an admin user with e-mail address set and:
+           a. account disabled & mail enabled
+           b. account enabled & mail disabled
+
+    :steps:
+        1. Create a long-running task.
+        3. For each user, wait and check that the notification e-mail has NOT been sent.
+
+    :BZ: 2245056
+
+    :customerscenario: true
+    """
+    assert admin_user_with_custom_settings
+    task_id = long_running_task['task']['id']
+    assert task_id
+
+    for email in root_mailbox_copy:
+        assert (
+            task_id not in email.as_string()
+        ), f'Unexpected notification e-mail with long-running task ID {task_id} found in user mailbox!'
```

Comment:
IMO this gives the test more "believable" look since nothing much else looks to be happening here :wink: 
And it's also a double safety, because if this assert fails, it means that the unwanted mail somehow slipped through `wait_for_no_long_running_task_mail` fixture (meaning it's not working as it should).